stdlib/public/core/StringBridge.swift

```diff
@@ -635,6 +647,23 @@ internal func _bridgeCocoaString(_ cocoaString: _CocoaString) -> _StringGuts {
 }
 
 extension String {
+  @_spi(Foundation)
+  public init<Encoding: Unicode.Encoding>(
+    _immortalCocoaString: AnyObject,
+    buffer: UnsafeBufferPointer<UInt8>,
+    encoding: Encoding.Type) {
+      //TODO: validate isa pointer, and use constant string isa pointer elsewhere
```

Comment:
I didn't want to mess with the code in CoreFoundationShims.h to add __NSCFConstantString's isa pointer for two reasons: 
1) Foundation already has it, and can decide when to call this as appropriate
2) it would cause merge conflicts currently

Possibly I should remove the buffer argument and just take the count. I think that would save one call on the Foundation side and we don't need the buffer ourselves right now.

Reply:
Update: I removed the buffer argument. I checked on the Foundation side and we would have needed an extra call to get to it.